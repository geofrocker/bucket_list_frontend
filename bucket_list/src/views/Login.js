import React, {Component} from 'react';
import {Link} from 'react-router-dom';

class Login extends Component{
    render(){
        return(
            <div className="auth">
                <div className="auth-container">
                    <div className="card">
                        <header className="auth-header">
                            <h1 className="auth-title">
                                BUCKET-LIST
                            </h1>
                        </header>
                        <div className="auth-content">
                            <p className="text-xs-center">LOGIN TO CONTINUE</p>
                            <form id="login-form" method="POST" onsubmit="return false">
                                <div className="form-group">
                                    <label for="email">Email Address</label>
                                    <input type="text" className="form-control underlined" name="email" id="email" placeholder="Your Email address" />
                                </div>
                                <div className="form-group">
                                    <label for="password">Password</label>
                                    <input type="password" className="form-control underlined" name="password" id="password" placeholder="Your Password" />
                                </div>

                                <div className="form-group">
                                    <button type="submit" className="btn btn-block btn-primary">Login</button>
                                </div>
                                <div className="form-group">
                                    <p className="text-muted text-xs-center">Do not have an account? <Link to="/register">Register</Link>
                                    </p>

                                </div>
                            </form>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}
export default Login;