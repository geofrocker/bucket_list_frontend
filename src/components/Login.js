import React, {Component} from 'react';
import {Link, Redirect } from 'react-router-dom';
import axios from 'axios';
import swal from 'sweetalert';

class Login extends Component{
    /**
     * This component is used to Login a user.
     * It handles the following methods:
     *  handleEmailChange() --> Used to set the state that is being changed in the value of the input to the state
     *  handlePasswordChange() --> Used to set the state of the password that is being changed
     *  handleSubmit() --> Used to submit data to the specified endpoint
     *  render --> Used to return jsx
     */

    constructor(props) {
        super(props);
        this.state = {email: '', password: '', redirect: false, token: ''};
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleEmailChange = this.handleEmailChange.bind(this);
        this.handlePasswordChange = this.handlePasswordChange.bind(this);
    }

    handleEmailChange(event){
        this.setState({email: event.target.value})
    }

    handlePasswordChange(event) {
        this.setState({password: event.target.value})
    }

    handleSubmit(event){
        event.preventDefault();
        axios({
            url: 'https://ridge-bucket-list-api.herokuapp.com/api/v1/auth/login',
            data: {email: this.state.email, password: this.state.password},
            datatype: "json",
            method: "post",
        })
            .then((response)=>{
                console.log(JSON.stringify(response));
                window.localStorage.setItem('token', response.data.token);
                window.localStorage.setItem("isLoggedIn", true);
                this.setState({redirect:true});
        })
            .catch((xhr) =>{
                swal("Error!", xhr.response.data.error, 'error');

            });
    }

    render(){
        if (this.state.redirect === true){
            return <Redirect to="/bucketlists/create"/>
        }
        return(
            <div className="auth">
                <div className="auth-container">
                    <div className="card">
                        <header className="auth-header">
                            <h1 className="auth-title">
                                BUCKET-LIST
                            </h1>
                        </header>
                        <div className="auth-content">
                            <p className="text-xs-center">LOGIN TO CONTINUE</p>
                            <form id="login-form" method="POST" onSubmit={this.handleSubmit}>
                                <div className="form-group">
                                    <label>Email Address</label>
                                    <input type="text" className="form-control underlined"  value={this.state.email} onChange={this.handleEmailChange} name="email" id="email" placeholder="Your Email address" />
                                </div>
                                <div className="form-group">
                                    <label>Password</label>
                                    <input type="password" className="form-control underlined" value={this.state.password} onChange={this.handlePasswordChange} name="password" id="password" placeholder="Your Password" />
                                </div>

                                <div className="form-group">
                                    <button type="submit" id="login-user" className="btn btn-block btn-primary">Login</button>
                                </div>
                                <div className="form-group">
                                    <p className="text-muted text-xs-center">Do not have an account? <Link to="/register">Register</Link>
                                    </p>

                                </div>
                            </form>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}
export default Login;