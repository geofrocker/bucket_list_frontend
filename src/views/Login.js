import React, {Component} from 'react';
import {Link, Redirect } from 'react-router-dom';
import axios from 'axios';
import swal from 'sweetalert';

class Login extends Component{

    constructor(props) {
        super(props);
        this.state = {email: '', password: '', redirect: false, token: ''};
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleEmailChange = this.handleEmailChange.bind(this);
        this.handlePasswordChange = this.handlePasswordChange.bind(this);
    }

    handleEmailChange(event){
        this.setState({email: event.target.value})
    }

    handlePasswordChange(event) {
        this.setState({password: event.target.value})
    }

    handleSubmit(event){
        event.preventDefault();
        axios({
            url: 'http://127.0.0.1:5000/api/v1/auth/login',
            data: {email: this.state.email, password: this.state.password},
            datatype: "json",
            method: "post"
        })
            .then((response)=>{
                console.log(JSON.stringify(response));
                window.localStorage.setItem('token', response.data.token);
                window.localStorage.setItem("isLoggedIn", true);
                this.setState({redirect:true});
        })
            .catch((xhr) =>{
                swal("Error!", xhr.response.data.error, 'error');

            });
    }

    render(){
        if (this.state.redirect === true){
            return <Redirect to="/bucketlists/create"/>
        }
        return(
            <div className="auth">
                <div className="auth-container">
                    <div className="card">
                        <header className="auth-header">
                            <h1 className="auth-title">
                                BUCKET-LIST
                            </h1>
                        </header>
                        <div className="auth-content">
                            <p className="text-xs-center">LOGIN TO CONTINUE</p>
                            <form id="login-form" method="POST" onSubmit={this.handleSubmit}>
                                <div className="form-group">
                                    <label>Email Address</label>
                                    <input type="text" className="form-control underlined"  value={this.state.email} onChange={this.handleEmailChange} name="email" id="email" placeholder="Your Email address" />
                                </div>
                                <div className="form-group">
                                    <label>Password</label>
                                    <input type="password" className="form-control underlined" value={this.state.password} onChange={this.handlePasswordChange} name="password" id="password" placeholder="Your Password" />
                                </div>

                                <div className="form-group">
                                    <button type="submit" id="login-user" className="btn btn-block btn-primary">Login</button>
                                </div>
                                <div className="form-group">
                                    <p className="text-muted text-xs-center">Do not have an account? <Link to="/register">Register</Link>
                                    </p>

                                </div>
                            </form>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}
export default Login;