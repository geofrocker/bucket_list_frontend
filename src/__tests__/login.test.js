import React from 'react';
import Login from '../components/Login';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import {MemoryRouter} from 'react-router-dom';

describe('<Login />', ()=>{
    it('contains form', function () {
        const wrapper = mount(
            <MemoryRouter>
            <Login/>
        </MemoryRouter>);
        const preventDefault = jest.fn();

        wrapper.setState({email:'ridgekimani@gmail.com', password:"milimani"});
        expect(wrapper.find("#login-form").length === 1);
        wrapper.find("#login-form").simulate('submit', {preventDefault});
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(preventDefault).toBeCalled();
    });

    it('should change the values to the email input', function(){
        const wrapper = mount(
            <MemoryRouter>
                <Login/>
            </MemoryRouter>);
        wrapper.find("#email").simulate('change', {target: {value: 'example@email.com'}});
        expect(toJson(wrapper)).toMatchSnapshot();

    });

    it('should change the values to the password input', function(){
        const wrapper = mount(
            <MemoryRouter>
                <Login/>
            </MemoryRouter>);
        wrapper.find("#password").simulate('change', {target: {value: 'milimani'}});
        expect(toJson(wrapper)).toMatchSnapshot();

    })

});
