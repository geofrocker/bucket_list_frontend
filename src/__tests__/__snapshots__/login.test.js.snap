// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> contains form 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "987wv9",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "987wv9",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Login>
      <div
        className="auth"
      >
        <div
          className="auth-container"
        >
          <div
            className="card"
          >
            <header
              className="auth-header"
            >
              <h1
                className="auth-title"
              >
                BUCKET-LIST
              </h1>
            </header>
            <div
              className="auth-content"
            >
              <p
                className="text-xs-center"
              >
                LOGIN TO CONTINUE
              </p>
              <form
                id="login-form"
                method="POST"
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <label>
                    Email Address
                  </label>
                  <input
                    className="form-control underlined"
                    id="email"
                    name="email"
                    onChange={[Function]}
                    placeholder="Your Email address"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <label>
                    Password
                  </label>
                  <input
                    className="form-control underlined"
                    id="password"
                    name="password"
                    onChange={[Function]}
                    placeholder="Your Password"
                    type="password"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <button
                    className="btn btn-block btn-primary"
                    id="login-user"
                    type="submit"
                  >
                    Login
                  </button>
                </div>
                <div
                  className="form-group"
                >
                  <p
                    className="text-muted text-xs-center"
                  >
                    Do not have an account? 
                    <Link
                      replace={false}
                      to="/register"
                    >
                      <a
                        href="/register"
                        onClick={[Function]}
                      >
                        Register
                      </a>
                    </Link>
                  </p>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </Login>
  </Router>
</MemoryRouter>
`;

exports[`<Login /> should change the values to the email input 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "37twil",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "37twil",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Login>
      <div
        className="auth"
      >
        <div
          className="auth-container"
        >
          <div
            className="card"
          >
            <header
              className="auth-header"
            >
              <h1
                className="auth-title"
              >
                BUCKET-LIST
              </h1>
            </header>
            <div
              className="auth-content"
            >
              <p
                className="text-xs-center"
              >
                LOGIN TO CONTINUE
              </p>
              <form
                id="login-form"
                method="POST"
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <label>
                    Email Address
                  </label>
                  <input
                    className="form-control underlined"
                    id="email"
                    name="email"
                    onChange={[Function]}
                    placeholder="Your Email address"
                    type="text"
                    value="example@email.com"
                  />
                </div>
                <div
                  className="form-group"
                >
                  <label>
                    Password
                  </label>
                  <input
                    className="form-control underlined"
                    id="password"
                    name="password"
                    onChange={[Function]}
                    placeholder="Your Password"
                    type="password"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <button
                    className="btn btn-block btn-primary"
                    id="login-user"
                    type="submit"
                  >
                    Login
                  </button>
                </div>
                <div
                  className="form-group"
                >
                  <p
                    className="text-muted text-xs-center"
                  >
                    Do not have an account? 
                    <Link
                      replace={false}
                      to="/register"
                    >
                      <a
                        href="/register"
                        onClick={[Function]}
                      >
                        Register
                      </a>
                    </Link>
                  </p>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </Login>
  </Router>
</MemoryRouter>
`;

exports[`<Login /> should change the values to the password input 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "m05ih0",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "m05ih0",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Login>
      <div
        className="auth"
      >
        <div
          className="auth-container"
        >
          <div
            className="card"
          >
            <header
              className="auth-header"
            >
              <h1
                className="auth-title"
              >
                BUCKET-LIST
              </h1>
            </header>
            <div
              className="auth-content"
            >
              <p
                className="text-xs-center"
              >
                LOGIN TO CONTINUE
              </p>
              <form
                id="login-form"
                method="POST"
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <label>
                    Email Address
                  </label>
                  <input
                    className="form-control underlined"
                    id="email"
                    name="email"
                    onChange={[Function]}
                    placeholder="Your Email address"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <label>
                    Password
                  </label>
                  <input
                    className="form-control underlined"
                    id="password"
                    name="password"
                    onChange={[Function]}
                    placeholder="Your Password"
                    type="password"
                    value="milimani"
                  />
                </div>
                <div
                  className="form-group"
                >
                  <button
                    className="btn btn-block btn-primary"
                    id="login-user"
                    type="submit"
                  >
                    Login
                  </button>
                </div>
                <div
                  className="form-group"
                >
                  <p
                    className="text-muted text-xs-center"
                  >
                    Do not have an account? 
                    <Link
                      replace={false}
                      to="/register"
                    >
                      <a
                        href="/register"
                        onClick={[Function]}
                      >
                        Register
                      </a>
                    </Link>
                  </p>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </Login>
  </Router>
</MemoryRouter>
`;
